packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-preview.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  css,
+  SpinLoader,
+  DocumentIcon,
+  spacing,
+  Overline,
+} from '@mongodb-js/compass-components';
+import type { RootState } from '../../../modules';
+import type { Document } from 'mongodb';
+import { DocumentListView } from '@mongodb-js/compass-crud';
+import HadronDocument from 'hadron-document';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  height: '100%',
+});
+
+const previewHeaderStyles = css({
+  paddingTop: spacing[3],
+  paddingBottom: spacing[3],
+  paddingLeft: spacing[3],
+});
+
+const centerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  height: '100%',
+});
+
+const messageStyles = css({ marginTop: spacing[3] });
+
+const documentListStyles = css({
+  overflow: 'scroll',
+});
+
+type PipelinePreviewProps = {
+  isLoading: boolean;
+  previewDocs: Document[] | null;
+};
+
+const PreviewResults = ({
+  previewDocs,
+  isLoading,
+}: {
+  previewDocs: Document[] | null;
+  isLoading: boolean;
+}) => {
+  const listProps: React.ComponentProps<typeof DocumentListView> = useMemo(
+    () => ({
+      docs: (previewDocs ?? []).map((doc) => new HadronDocument(doc)),
+      isEditable: false,
+      copyToClipboard(doc) {
+        const str = doc.toEJSON();
+        void navigator.clipboard.writeText(str);
+      },
+    }),
+    [previewDocs]
+  );
+
+  if (isLoading) {
+    return (
+      <div className={centerStyles}>
+        <SpinLoader size="24px" />
+      </div>
+    );
+  }
+
+  if (!previewDocs) {
+    return (
+      <div className={centerStyles}>
+        <DocumentIcon size={spacing[6]} />
+        <Body className={messageStyles}>
+          Preview results to see a sample of the aggregated results from this
+          pipeline.
+        </Body>
+      </div>
+    );
+  }
+
+  if (previewDocs.length === 0) {
+    return (
+      <div className={centerStyles}>
+        <DocumentIcon size={spacing[6]} />
+        <Body className={messageStyles}>No preview documents</Body>
+      </div>
+    );
+  }
+
+  return (
+    <div className={documentListStyles}>
+      <DocumentListView {...listProps} />
+    </div>
+  );
+};
+
+export const PipelinePreview: React.FunctionComponent<PipelinePreviewProps> = ({
+  isLoading,
+  previewDocs,
+}) => {
+  const docCount = previewDocs?.length;
+  const docText = docCount === 1 ? 'document' : 'documents';
+  return (
+    <div className={containerStyles} data-testid="pipeline-as-text-preview">
+      <div className={previewHeaderStyles}>
+        <Overline>Pipeline Output</Overline>
+        {!isLoading && docCount ? (
+          <Body>{`Sample of ${docCount} ${docText}`}</Body>
+        ) : (
+          <></>
+        )}
+      </div>
+      <PreviewResults isLoading={isLoading} previewDocs={previewDocs} />
+    </div>
+  );
```

Comment:
Tiny nit: you don't really need to return empty fragments if you don't want to render anything:

```tsx
        {!isLoading && docCount && (
          <Body>{`Sample of ${docCount} ${docText}`}</Body>
        )}
```

Additionally as 0 is falsy you are running into corner cases here with the way the code structured. What I'd suggest you to do here is to move the condition into a correctly typed variable just so it's clear what's the expectation for what you are trying to achieve here:

```suggestion
  const docCount = previewDocs?.length ?? 0;
  const shouldShowCount = !isLoading && docCount > 0 // or don't check docCount if we want to render always even when count is zero
  const docText = docCount === 1 ? 'document' : 'documents';
  return (
    <div className={containerStyles} data-testid="pipeline-as-text-preview">
      <div className={previewHeaderStyles}>
        <Overline>Pipeline Output</Overline>
        {shouldShowCount && (
          <Body>{`Sample of ${docCount} ${docText}`}</Body>
        )}
      </div>
      <PreviewResults isLoading={isLoading} previewDocs={previewDocs} />
    </div>
  );
```

That way whoever is working with this code in the future knows exactly what's the intention here was